pandas/core/strings.py

```diff
@@ -1962,11 +1935,298 @@ def cons_row(x):
                 cons = self._orig._constructor
                 return cons(result, name=name, index=index)
 
-    @copy(str_cat)
-    def cat(self, others=None, sep=None, na_rep=None):
-        data = self._orig if self._is_categorical else self._data
-        result = str_cat(data, others=others, sep=sep, na_rep=na_rep)
-        return self._wrap_result(result, use_codes=(not self._is_categorical))
+    def _get_series_list(self, others, ignore_index=False):
+        """
+        Auxiliary function for :meth:`str.cat`. Turn potentially mixed input
+        into a list of Series (elements without an index must match the length
+        of the calling Series/Index).
+
+        Parameters
+        ----------
+        input : Series, DataFrame, np.ndarray, list-like or list-like of
+            objects that are either Series, np.ndarray (1-dim) or list-like
+        ignore_index : boolean, default False
+            Determines whether to forcefully align with index of the caller
+
+        Returns
+        -------
+        tuple : (input transformed into list of Series,
+                 Boolean whether FutureWarning should be raised)
+        """
+
+        # once str.cat defaults to alignment, this function can be simplified;
+        # will not need `ignore_index` and the second boolean output anymore
+
+        from pandas import Index, Series, DataFrame, isnull
+
+        # self._orig is either Series or Index
+        idx = self._orig if isinstance(self._orig, Index) else self._orig.index
+
+        err_msg = ('others must be Series, Index, DataFrame, np.ndarrary or '
+                   'list-like (either containing only strings or containing '
+                   'only objects of type Series/Index/list-like/np.ndarray)')
+
+        if isinstance(others, Series):
+            fu_wrn = not others.index.equals(idx)
+            los = [Series(others.values, index=idx)
+                   if ignore_index and fu_wrn else others]
+            return (los, fu_wrn)
+        elif isinstance(others, Index):
+            fu_wrn = not others.equals(idx)
+            los = [Series(others.values,
+                          index=(idx if ignore_index else others))]
+            return (los, fu_wrn)
+        elif isinstance(others, DataFrame):
+            fu_wrn = not others.index.equals(idx)
+            if ignore_index and fu_wrn:
+                # without copy, this could change "others"
+                # that was passed to str.cat
+                others = others.copy()
+                others.index = idx
+            return ([others[x] for x in others], fu_wrn)
+        elif isinstance(others, np.ndarray) and others.ndim == 2:
```

Comment:
this is wrong

i don’t think we can align a ndarray at all like this
let’s can ndarray a that are > 1 dim

Reply:
The DF-constructor works as expected for a 2-dim ndarray, but I haven't checked if this is tested behaviour. (essentially, `df == DataFrame(df.values, columns=df.columns, index=df.index)`)

I would suggest not to can 2-dim ndarrays, because they are necessary to avoid alignment on the deprecation path for `join`:
> [...] To disable alignment (the behavior before v.0.23) and  silence this warning, use `.values` on any Series/Index/DataFrame in `others`. [...]